alembic/versions/20230531_0af587ff8595_migrate_license_integrations_to_.py

```diff
@@ -0,0 +1,106 @@
+"""Migrate license integrations to configuration settings
+
+Revision ID: 0af587ff8595
+Revises: b883671b7bc5
+Create Date: 2023-05-31 12:34:42.550703+00:00
+
+"""
+
+from typing import Type
+
+from alembic import op
+from api.integration.registry.license_providers import LicenseProvidersRegistry
+from core.integration.base import HasLibraryIntegrationConfiguration
+from core.integration.settings import BaseSettings
+from core.migration.migrate_external_integration import (
+    _migrate_external_integration,
+    _migrate_library_settings,
+    get_configuration_settings,
+    get_integrations,
+    get_library_for_integration,
+)
+
+# revision identifiers, used by Alembic.
+revision = "0af587ff8595"
+down_revision = "b883671b7bc5"
+branch_labels = None
+depends_on = None
+
+
+LICENSE_GOAL = "LICENSE_GOAL"
+
+
+def upgrade() -> None:
+    registry = LicenseProvidersRegistry()
+
+    connection = op.get_bind()
+
+    # Fetch all license type integrations
+    # The old enum had 'licenses', the new enum has 'LICENSE_GOAL'
+    integrations = get_integrations(connection, "licenses")
+    for integration in integrations:
+        _id, protocol, name = integration
+
+        # Get the right API class for it
+        api_class = registry.get(protocol, None)
+        if not api_class:
+            raise RuntimeError(f"Could not find API class for '{protocol}'")
+
+        # Create the settings and library settings dicts from the configurationsettings
+        settings_dict, library_settings, self_test_result = get_configuration_settings(
+            connection, integration
+        )
+
+        if "IGNORED_IDENTIFIER_TYPE" in settings_dict.keys():
+            value = settings_dict.pop("IGNORED_IDENTIFIER_TYPE")
+            settings_dict["ignored_identifier_types"] = value
```

Comment:
I'll rename the others but I'd like to change this to the lowercase scheme so the data is more consistent in the DB

Reply:
If you set this as an alias, the data does get stored in the DB as the name of the key. The alias is only used when the data is loaded into the Pydantic model. If the key is missing, Pydantic checks the alias and loads it as the key.

Reply:
Currently in the ConfigurationSettings the key name is uppercase, so in the data-driven approach it is a one to one migration for key naming. In order to move the name to lowercase in the DB I have to have this code in place.

Reply:
I don't understand what you are saying here @RishiDiwanTT. I agree this should be lowercase, but I don't understand why this can't be handled using an alias, like the rest of the settings that need to have their names changed as part of this. Pydantic is case sensitive, so it doesn't seem that we need a special path for this name change.

Reply:
Ok, I was running with some misconceptions about alias, I've cleared those out now :+1: